Tests/ZIPFoundationTests/ZIPFoundationWritingTests+Zip64.swift

```diff
@@ -0,0 +1,316 @@
+            XCTAssertEqual(cdData.scanValue(start: relativeCDExtraFieldOffset + 4), Int(dataSize))
+            XCTAssertEqual(cdData.scanValue(start: relativeCDExtraFieldOffset + 12), Int(dataSize))
+            return cdSize
+        } catch {
+            XCTFail("Unexpected error while reading chunk from archive file.")
+        }
+        return 0
+    }
+
+    private func checkZip64EndOfCentralDirectory(archive: Archive, cdSize: Int, cdOffset: Int, zip64EOCDOffset: Int,
+                                                 readData: (Int) throws -> Data) -> Int {
+        XCTAssertEqual(archive.endOfCentralDirectoryRecord.offsetToStartOfCentralDirectory, UInt32.max)
+        XCTAssertEqual(Int(archive.zip64EndOfCentralDirectory?.record.offsetToStartOfCentralDirectory ?? 0), cdOffset)
+        do {
+            let zip64EOCDSize = 56 + 20
+            let zip64EOCDData = try readData(zip64EOCDSize)
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 0), UInt32(zip64EOCDRecordStructSignature))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 4), UInt(44))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 12), UInt16(789))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 14), zip64Version)
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 16), UInt32(0))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 20), UInt32(0))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 24), UInt(1))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 32), UInt(1))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 40), Int(cdSize))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 48), Int(cdOffset))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 56), UInt32(zip64EOCDLocatorStructSignature))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 60), UInt32(0))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 64), Int(zip64EOCDOffset))
+            XCTAssertEqual(zip64EOCDData.scanValue(start: 72), UInt32(1))
+            return zip64EOCDSize
+        } catch {
+            XCTFail("Unexpected error while reading chunk from archive file.")
+        }
+        return 0
+    }
+
+    /// Target fields: Relative Offset of Local Header
+    func testAddEntryToArchiveWithZip64LFHOffset() {
+        mockIntMaxValues()
+        defer { resetIntMaxValues() }
+        let archive = self.archive(for: #function, mode: .update)
+        let size = 64 * 64 * 2
+        let data = Data.makeRandomData(size: size)
+        let entryName = ProcessInfo.processInfo.globallyUniqueString
+//        let currentLFHOffset = archive.offsetToStartOfCentralDirectory
+        do {
+            try archive.addFileEntry(with: entryName, size: size, data: data)
+        } catch {
+            XCTFail("Failed to add zip64 format entry to archive with error : \(error)"); return
+        }
+        // Please ignore this test for now, will be fixed later
```

Comment:
Can this be un-commented?

Reply:
Currently the answer is no 😢 , `readChunk(..)` is called in `makeIterator(..)`, thus these tests will be fixed in next step.